test/integration/util/util.go

```diff
@@ -68,13 +68,7 @@ type ShutdownFunc func()
 
 // StartApiserver starts a local API server for testing and returns the handle to the URL and the shutdown function to stop it.
 func StartApiserver() (string, ShutdownFunc) {
-	h := &framework.APIServerHolder{Initialized: make(chan struct{})}
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		<-h.Initialized
-		h.M.GenericAPIServer.Handler.ServeHTTP(w, req)
-	}))
-
-	_, _, closeFn := framework.RunAnAPIServerUsingServer(framework.NewIntegrationTestControlPlaneConfig(), s, h)
+	_, s, closeFn := framework.RunAnAPIServer(framework.NewIntegrationTestControlPlaneConfig())
```

Comment:
the CloseFn returned already has a s.Close()
https://github.com/kubernetes/kubernetes/blob/ce50018a760dda229377f29032492cfee94e52ec/test/integration/framework/master_utils.go#L127-L153

I think that the `shutdownFunc` can be removed with this change

Reply:
Good catch - PTAL